### ISSUE TITLE
Notify user when another one has network issues

### ISSUE THREAD

**Issue by lexin** [VoIPGRID/VialerSIPLib#146]
User A calls user B. Call starts. If A turns if Internet for a while (lets say 10 sec) the connection will be alive. Just A and B hear nothing. it's useful thing, because when Internet connection be available again users are able to continue without any additional actions. But how no notify A and B that it's just temporary issue with network? What I want to have it's some ring instead if silence and also the text message. But I can't find how to handle this case. Is it posssible?

**Comment by redmerloen** [comment]
A watcher on if there is actually audio is a good idea. I think the library will post a notification when there is no audio. And that your implementation of this library should handle it if the app posts the notification of no audio.

**Comment by lexin** [comment]
Thanks. I'll try to find this notification about no audio. Not sure that it exists.

**Comment by redmerloen** [comment]
This library is going be throwing that notification. I'm currently writing the code for that.

**Comment by lexin** [comment]
Ok. So I need to wait for a new version?

**Comment by redmerloen** [comment]
Yes for this feature you should wait

**Comment by lexin** [comment]
I am trying to use VSLCallNoAudioForCallNotification. And getting VSLCallAudioStateNoAudioReceiving in case if user on the other side loses Internet connection or turns on Hold. So, it's no possible to know - should we already end call because the reason is an absence of connection  or just wait because call just on hold.

**Comment by lexin** [comment]
Is it possible to detect that other user turns on Holding call?

**Comment by redmerloen** [comment]
If the remote party puts the user on hold. the media state should be `VSLMediaStateRemoteHold`.
I would not terminate the call immediately. You can inform user that there is no audio receiving.

**Comment by lexin** [comment]
I am trying to check it for example in 
- (void)checkIfAudioPresent - just for test. 
self.mediaState always PJSUA_CALL_MEDIA_ACTIVE (== 1) even remote users puts call on hold . 
It changes only to PJSUA_CALL_MEDIA_LOCAL_HOLD when I turn on holding in my side.

**Comment by lexin** [comment]
Also it seems that mediaStateChanged catches only after changes on my side  - not after remote actions (putting on hold etc.)

**Comment by redmerloen** [comment]
Okay will check later if i it's possible to fix the REMOTE_HOLD check

**Comment by lexin** [comment]
Hi. Did you have a chance to check that issue? I tried one more time, but issue is the same.

**Comment by redmerloen** [comment]
REMOTE_HOLD can only work if the endpoint a PBX sents the package that the call was put onhold on that side, i don't think it can work otherwise.

**Comment by lexin** [comment]
Ok. I see. Thanks. Try to handle it.
